include/openPMD/backend/Attributable.hpp

```diff
@@ -410,11 +433,28 @@ OPENPMD_protected
     virtual void linkHierarchy(Writable &w);
 }; // Attributable
 
-// TODO explicitly instantiate Attributable::setAttribute for all T in Datatype
 template <typename T>
 inline bool Attributable::setAttribute(std::string const &key, T value)
 {
-    internal::attr_value_check(key, value);
+    return setAttributeImpl(
+        key, std::move(value), internal::SetAttributeMode::FromPublicAPICall);
+}
+
+inline bool
+Attributable::setAttribute(std::string const &key, char const value[])
+{
+    return setAttributeImpl(
+        key, value, internal::SetAttributeMode::FromPublicAPICall);
+}
+
+// TODO explicitly instantiate Attributable::setAttribute for all T in Datatype
```

Comment:
This is done in `src/backend/Attributable.cpp`
```suggestion
// note: we explicitly instantiate Attributable::setAttributeImpl for all T in Datatype in Attributable.cpp
```

Reply:
added to 2nd commit.